### PR TITLE
Harden SignatureEquals nullability checks

### LINKED ISSUE #52492: Harden nullability checks in `Endpoint.SignatureEquals`
Logs indicate that unhandled NREs are thrown from the `Endpoint.SignatureEquals` from the `RequestDelegateGenerator` under certain and rare occasions. Although we've not be able to identify what the impacted syntax is, we should try to harden the nullability checks here to avoid situations where they do happen.

### PR DESCRIPTION
**Pull request by captainsafia**
Closes https://github.com/dotnet/aspnetcore/issues/52492

### CODE REVIEW

File: src/Http/Http.Extensions/gen/StaticRouteHandlerModel/Endpoint.cs [hunk 1]
```diff
@@ -108,8 +108,13 @@ public Endpoint(IInvocationOperation operation, WellKnownTypes wellKnownTypes, S
 
     public static bool SignatureEquals(Endpoint a, Endpoint b)
```
Comment (JamesNK) [inline src/Http/Http.Extensions/gen/StaticRouteHandlerModel/Endpoint.cs:109]: Should the parameters be nullable?
```suggestion
    public static bool SignatureEquals(Endpoint? a, Endpoint? b)
```

File: src/Http/Http.Extensions/gen/StaticRouteHandlerModel/Endpoint.cs [hunk 2]
```diff
@@ -108,8 +108,13 @@ public Endpoint(IInvocationOperation operation, WellKnownTypes wellKnownTypes, S
...
     {
+        if (a == null || b == null)
```
Comment (JamesNK) [inline src/Http/Http.Extensions/gen/StaticRouteHandlerModel/Endpoint.cs:111]: Handle both being null?

Comment (JamesNK) [inline src/Http/Http.Extensions/gen/StaticRouteHandlerModel/Endpoint.cs:111]:
```suggestion
        if (a == null && b == null)
        {
            return true;
        }

        if (a == null || b == null)
```

**Comment by JamesNK** [comment]
Unit tests?

File: src/Http/Http.Extensions/gen/StaticRouteHandlerModel/Endpoint.cs (hunk 2 above)
Comment (amcasey) [reply src/Http/Http.Extensions/gen/StaticRouteHandlerModel/Endpoint.cs:111]: Could do so by returning `ReferenceEquals(a, b)`

**Pull request closed without merging**